lib/features/composer/domain/usecases/create_new_and_save_email_to_drafts_interactor.dart

```diff
@@ -43,54 +48,78 @@ class CreateNewAndSaveEmailToDraftsInteractor {
-              emailDraftSaved.id!,
-              currentMailboxState: listCurrentState?.value1,
-              currentEmailState: listCurrentState?.value2
-            )
-          );
-        } else {
-          yield dartz.Right<Failure, Success>(UpdatingEmailDrafts());
-
-          final emailDraftSaved = await _emailRepository.updateEmailDrafts(
-            createEmailRequest.session,
-            createEmailRequest.accountId,
-            emailCreated,
-            createEmailRequest.draftsEmailId!,
-            cancelToken: cancelToken
-          );
-
-          yield dartz.Right<Failure, Success>(
-            UpdateEmailDraftsSuccess(
-              emailDraftSaved.id!,
-              currentMailboxState: listCurrentState?.value1,
-              currentEmailState: listCurrentState?.value2
-            )
-          );
+      if (emailCreated == null) {
+        yield dartz.Left<Failure, Success>(GenerateEmailFailure(CannotCreateEmailObjectException()));
+        return;
+      }
+
+      if (enableTimeout) {
+        _timeoutInterceptors.setTimeout(
+          connectionTimeout: AppConfig.savingMessageTimeout,
+          sendTimeout: AppConfig.savingMessageTimeout,
+          receiveTimeout: AppConfig.savingMessageTimeout,
+        );
+      }
+
+      if (createEmailRequest.draftsEmailId == null) {
+        yield dartz.Right<Failure, Success>(SaveEmailAsDraftsLoading());
+
+        final emailDraftSaved = await _emailRepository.saveEmailAsDrafts(
+          createEmailRequest.session,
+          createEmailRequest.accountId,
+          emailCreated,
+          cancelToken: cancelToken
+        );
+
+        if (enableTimeout) {
+          _timeoutInterceptors.resetTimeout();
         }
+
+        yield dartz.Right<Failure, Success>(
+          SaveEmailAsDraftsSuccess(
+            emailDraftSaved.id!,
+            currentMailboxState: listCurrentState?.value1,
+            currentEmailState: listCurrentState?.value2
+          )
+        );
       } else {
-        yield dartz.Left<Failure, Success>(GenerateEmailFailure(CannotCreateEmailObjectException()));
+        yield dartz.Right<Failure, Success>(UpdatingEmailDrafts());
+
+        final emailDraftSaved = await _emailRepository.updateEmailDrafts(
+          createEmailRequest.session,
+          createEmailRequest.accountId,
+          emailCreated,
+          createEmailRequest.draftsEmailId!,
+          cancelToken: cancelToken
+        );
+
+        if (enableTimeout) {
+          _timeoutInterceptors.resetTimeout();
+        }
+
+        yield dartz.Right<Failure, Success>(
+          UpdateEmailDraftsSuccess(
+            emailDraftSaved.id!,
+            currentMailboxState: listCurrentState?.value1,
+            currentEmailState: listCurrentState?.value2
+          )
+        );
       }
     } catch (e) {
       logError('CreateNewAndSaveEmailToDraftsInteractor::execute: Exception: $e');
+      if (enableTimeout) {
+        _timeoutInterceptors.resetTimeout();
+      }
```

Comment:
- write unit test to make sure `ressetTimeout` will be invoked in all cases (success and failure)

Reply:
done